AiTableTopGameMaster.ConsoleApp/Clients/AgentChatClient.cs

```diff
@@ -0,0 +1,109 @@
+using AiTableTopGameMaster.ConsoleApp.Helpers;
+using JetBrains.Annotations;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Agents;
+using Microsoft.SemanticKernel.ChatCompletion;
+using Spectre.Console;
+
+namespace AiTableTopGameMaster.ConsoleApp.Clients;
+
+/// <summary>
+/// Agent-based chat client implementation using Semantic Kernel's agents framework.
+/// This replaces the direct IChatCompletionService approach with a more structured agent model.
+/// </summary>
+[UsedImplicitly]
+public class AgentChatClient(
```

Comment:
The logic in this class should really just be in ConsoleChatClient. I don't see a compelling reason for this to exist.